Sources/App/ThreadCredentialsSharing/ThreadCredentialsSharingView.swift

```diff
@@ -0,0 +1,134 @@
+                    Button {
+                        dismiss()
+                    } label: {
+                        Text(L10n.doneLabel)
+                    }
+                }
+            })
+            .alert(alertTitle, isPresented: $viewModel.showAlert) {
+                errorAlertActions
+            } message: {
+                if case let .error(_, message) = viewModel.alertType {
+                    Text(message)
+                }
+            }
+        }
+        .onAppear {
+            Task {
+                await viewModel.retrieveAllCredentials()
+            }
+        }
+    }
+
+    private var alertTitle: String {
+        if case let .error(title, _) = viewModel.alertType {
+            return title
+        } else if case let .success(title) = viewModel.alertType {
+            return title
+        } else {
+            return ""
+        }
+    }
+
+    @ViewBuilder
+    private var errorAlertActions: some View {
+        Button {
+            /* no-op */
+        } label: {
+            Text(L10n.doneLabel)
+        }
+
+        if case .error = viewModel.alertType {
+            Button {
+                Task {
+                    await viewModel.retrieveAllCredentials()
+                }
+            } label: {
+                Text(L10n.retryLabel)
+            }
+        }
+    }
+
+    private var progressView: some View {
+        ProgressView()
+            .progressViewStyle(.circular)
+            .scaleEffect(CGSize(width: 2, height: 2))
+    }
+
+    @ViewBuilder
+    private var credentialsList: some View {
+        if viewModel.credentials.isEmpty {
+            Text("You don't have credentials available on your iCloud Keychain.")
+                .multilineTextAlignment(.center)
+        } else {
+            List(viewModel.credentials, id: \.borderAgentID) { credential in
+                makeCredentialCard(credential: credential)
+                    .listRowSeparator(.hidden)
+                    .listRowBackground(Color.clear)
+            }
+            .listStyle(.plain)
+        }
+    }
+
+    private func makeCredentialCard(credential: ThreadCredential) -> some View {
+        CardView(backgroundColor: Color(uiColor: .systemBackground)) {
+            makeCardPropertyView(
+                title: L10n.Thread.Credentials.networkNameTitle,
+                value: credential.networkName
+            )
+            makeCardPropertyView(
+                title: L10n.Thread.Credentials.borderAgentIdTitle,
+                value: credential.borderAgentID
+            )
+            makeCardPropertyView(
+                title: L10n.Thread.Credentials.networkKeyTitle,
+                value: credential.networkKey
+            )
+            Button {
+                viewModel.shareCredentialWithHomeAssistant(credential: credential)
+            } label: {
+                Text(L10n.Thread.Credentials.shareCredentialsButtonTitle)
+            }
```

Comment:
can this button get a loading indicator while the network request is executing, or make it disabled?

can we get a success HUD of some sort, or alert?

Reply:
There is a success alert already, I won't change this for now because today I have a meeting with the designer so there is a chance more in terms of UI will be changed, but it can be an iteration after merging this PR